### PR TITLE
Return search term suggestions

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
- [x] Settings
  - [x] New settings are documented and present in `app.json`
  - [x] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
#2506 
#2507

#### What's this PR do?
- Removes fuzziness from course search
- Returns a suggested search term if the number of results is <= X (default 1)
- Refactors the search code, so that separate functions are called for learning resource search vs discussions search

#### How should this be manually tested?
- You should have a decent number of imported courses (like edx).
- Recreate the index (there are changes to the analyzers and mappings).
- Search for 'enginr', you should get 0 results back and "Did you mean engineer" should be displayed.
- Click on the "engineer" link, it should rerun the search with that term.
- Try again with "mechical enginr", the suggestion should be "mechanical engineer"
- Run searches on posts, comments, and profiles, they should work as before (and no suggestions).

#### Screenshots (if appropriate)
in comments below

### CODE REVIEW

**Comment by mbertrand** [comment]
@abdulkdawson @Ferdi here's a screenshot of what the UI currently looks like:

Screen Shot 2019-12-20 at 4 43 13 PM

**Comment by abdulkdawson** [comment]
@mbertrand this is starting out good, but the "did you mean" text should be above the grid and list icons.

**Comment by mbertrand** [comment]
@ferdi I've noticed a few issues in terms of how suggestions work:
- If I search for `mechanical enginr`, the suggestions won't appear because I get a sufficient number of results just for `mechanical` - once we support quoted phrase search I suspect this won't be an issue as long as users add the quotes.
- Elasticsearch's suggestions seem reasonable if I misspell one of 2+ words in a phrase, for example "mechanical enginr" -> "mechanical engineer".  However, if I misspell both words (like `mechical enginering`), the highest scored suggestions have 1 word or the other still misspelled (example below).  I'll experiment today with tweaking some of the query parameters to try to fix this.
```
                    {
                        "text": "mechical engineering",
                        "score": 0.10253522
                    },
                    {
                        "text": "mechanical engineering",
                        "score": 0.005519242
                    },
                    {
                        "text": "medical engineering",
                        "score": 0.0027011232
                    },
                    {
                        "text": "mechanical enginering",
                        "score": 0.0017430395
                    },
                    {
                        "text": "medical enginering",
                        "score": 0.0010172967
                    }
```

**Comment by mbertrand** [comment]
@abdulkdawson @Ferdi an updated screenshot:

Screen Shot 2019-12-26 at 12 39 38 PM


I also made the number of suggestions returned configurable in case we ever decide to show more than 1, here's what that looks like:

Screen Shot 2019-12-26 at 8 54 10 AM

Since quoted search isn't supported yet, you probably won't see suggestions for multiple words unless the search hit threshold to display them is set very high, or all words in the phrase are misspelled.  Here's one that works:

Screen Shot 2019-12-26 at 12 44 55 PM

**Comment by mbertrand** [comment]
@Ferdi one assumption I made is that suggestions are only for learning resource searches, not discussion (post, comment, profile) searches - is that correct?

File: search/views.py [hunk 1]
```diff
@@ -27,14 +33,18 @@ def handle_exception(self, exc):
 
 class SearchView(ESView):
     """
-    View for executing searches
+    View for executing searches of posts, comments, profiles
```
Comment (abeglova) [inline search/views.py]: ...and learning recourses?

**Comment by abeglova** [comment]
Works as advertised! Back to you with a few minor comments

**Comment by mbertrand** [comment]
Thanks @abeglova, the code has been updated

**Comment by abeglova** [comment]
LGTM ready to merge

**Comment by mbertrand** [comment]
@pdpinch @abdulkdawson latest screenshots are [here](https://github.com/mitodl/open-discussions/pull/2510#issuecomment-569101372)

**Comment by abdulkdawson** [comment]
@mbertrand This is looking good.

**Pull request merged**